04week/mastermind.js

```diff
@@ -28,13 +28,47 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(solution, guess) {
   // your code here
+  let solutionArray = solution.split('');
+  let guessArray = guess.split('');
+  let correctLetterLocations = 0;
+  let correctLetters = 0;
+
+for (var i = 0; i < solutionArray.length; i++) {
+  if (solutionArray[i] === guessArray[i]) {
+       correctLetterLocations++;
+       solutionArray[i] = null;
+     }
+   }
+
+for (var y = 0; y < solutionArray.length; y++) {
+  let targetIndex = solutionArray.indexOf(guessArray[y]);
+
+ if (targetIndex > -1) {
+      correctLetters++;
+      solutionArray[y] = null;
+    }
+  }
+  return colors.red(correctLetterLocations) + '-' + colors.white(correctLetters);
 }
 
+
 function mastermind(guess) {
   solution = 'abcd'; // Comment this out to generate a random solution
   // your code here
+  if (guess == solution){
+    return "You guessed it!"
```

Comment:
I think you were supposed to udpate the board, or end the game if they guessed the correct answer.